### PR TITLE
Consumable signatures

### PR DESCRIPTION
**Pull request by shargon**

### CODE REVIEW

**Comment by shargon** [comment]
Yes, we could have two syscalls, one `Runtime.ConsumeWitness`, i agree

**Comment by vncoelho** [comment]
@shargon, sounds good this discussion, I believe that this way of reasoning gonna bring us to something even more powerful, safer and flexible.

File: neo/SmartContract/ApplicationEngine.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public partial class ApplicationEngine : ExecutionEngine
         private readonly List<IDisposable> disposables = new List<IDisposable>();
 
         public TriggerType Trigger { get; }
-        public IVerifiable ScriptContainer { get; }
+        public ConsumableWitnesses ScriptContainer { get; }
```
Comment (igormcoelho) [inline neo/SmartContract/ApplicationEngine.cs]: I think that a better approach here, is having a vector of verified scripthashes. Lets keep only IVerifiable class,and we remove consumed witnesses from that list.
So we load the script and already pass the available validated witnesses.

**Comment by erikzhang** [comment]
https://github.com/neo-project/neo/blob/29fb4ed794d21e5f8015b6e00e158901ec2942e1/neo/Network/P2P/Payloads/TransactionAttributeUsage.cs#L5

The easiest way is to add a "valid number" field to `Cosigner`.

**Comment by vncoelho** [comment]
"valid number" will the max number of times that it can be used? 
Co-signer involve the signature of the sender?

**Comment by shargon** [comment]
I will close this because is better scoped signatures

**Pull request closed without merging**